lib/nimbase.h

```diff
@@ -310,29 +310,24 @@ __AVR__
 namespace USE_NIM_NAMESPACE {
 #endif
 
-// preexisting check, seems paranoid, maybe remove
-#if defined(NIM_TRUE) || defined(NIM_FALSE) || defined(NIM_BOOL)
-#error "nim reserved preprocessor macros clash"
-#endif
-
 /* bool types (C++ has it): */
-#ifdef __cplusplus
-#define NIM_BOOL bool
-#elif (defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901)
-// see #13798: to avoid conflicts for code emitting `#include <stdbool.h>`
-#define NIM_BOOL _Bool
-#else
-typedef unsigned char NIM_BOOL; // best effort
+#ifndef __cplusplus
+#  if (defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901)
+#    include <stdbool.h>
+#  else
+#    define bool  unsigned char
```

Comment:
This is confusingly not using the C bool header. I know that I told you not to use it, but NIM_TRUE and NIM_FALSE are not misleading to the casual reader. So let's keep them.

Reply:
First of all: thanks for having another look at this!

[If we _still_ want to support C89](https://github.com/nim-lang/Nim/pull/13798#issuecomment-607332944), we can't assume the presence of the bool header, which is part of the ISO C**9**9 spec, unfortunately (cf. https://en.wikibooks.org/wiki/C_Programming/stdbool.h, https://stackoverflow.com/a/1921557, https://stackoverflow.com/a/1608350 for a cursory glance).
The strategy taken by C89 programmers to work with some sort of fake bool type seems to be the one I take: typedefing `bool`, and defining `true`, and `false` as macros.
For a pretty pedantic C89 support, our current definitions are broken anyway, see https://github.com/nim-lang/Nim/blob/devel/lib/nimbase.h#L305-L321: we define `NIM_TRUE`/`NIM_FALSE` as `true`/`false`, but they don't exist in "pure" C89 (I guess merely as Compiler extensions/hacks).

Given that, do we actually still want to retain C89 support?

Reply:
`true` is C's `true` which maps to  `_Bool` or whatever it's called which is not necessarily of size 1. Hence we avoid `_Bool` and then our own `true` would be confusing.

Reply:
So I vote for NIM_TRUE/NIM_FALSE regardless of our increasingly broken C89 support.